treewide: remove paxutils

Mic92 opened the pull request:
More then one year ago we removed grsecurity kernels from nixpkgs:
https://github.com/NixOS/nixpkgs/pull/25277

This removes now also paxutils from stdenv.
I will move this to staging soon.

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

joachifm reviewed (approved):
I suppose we can infer that nobody cares about nixpkgs on grsec/pax ...

joachifm commented on pkgs/top-level/all-packages.nix:
```diff
@@ -15023,12 +15023,6 @@ in
 
-  paxctl = callPackage ../os-specific/linux/paxctl { };
-
-  paxtest = callPackage ../os-specific/linux/paxtest { };
```
I think that leaving at least paxtest and maybe even paxctl makes sense; the former can be used to assess whether vanilla mitigations are able top catch stuff covered by PaX, which I find useful (given the ongoing effort to backport/lift features from grsec/pax).

Mic92 committed:
treewide: remove paxutils from stdenv

More then one year ago we removed grsecurity kernels from nixpkgs:
https://github.com/NixOS/nixpkgs/pull/25277

This removes now also paxutils from stdenv.

Mic92 commented:
cherry-picked to 1b146a8c6f5
